src/main/scala/sbtavro/SbtAvro.scala

```diff
@@ -110,40 +177,51 @@ object SbtAvro extends AutoPlugin {
     compiler.compileToDestination(null, target)
   }
 
-  private[this] def compileAvroSchema(srcDir: File, target: File, log: Logger, stringTypeName: String, fieldVisibilityName: String, enableDecimalLogicalType: Boolean, useNamespace: Boolean): Set[File] = {
-    val stringType = StringType.valueOf(stringTypeName)
-    val fieldVisibility = SpecificCompiler.FieldVisibility.valueOf(fieldVisibilityName.toUpperCase)
-    log.info("Avro compiler using stringType=%s".format(stringType))
-
-    for (idl <- (srcDir ** AvroAvdlFilter).get) {
-      log.info("Compiling Avro IDL %s".format(idl))
+  private[this] def compileAvroSchema(srcDir: File,
+                                      target: File,
+                                      log: Logger,
+                                      stringType: StringType,
+                                      fieldVisibility: FieldVisibility,
+                                      enableDecimalLogicalType: Boolean,
+                                      useNamespace: Boolean): Set[File] = {
+    (srcDir ** AvroAvdlFilter).get.foreach { idl =>
+      log.info(s"Compiling Avro IDL $idl")
       compileIdl(idl, target, stringType, fieldVisibility, enableDecimalLogicalType)
     }
 
-    compileAvscs(srcDir, target, stringType, fieldVisibility, enableDecimalLogicalType, useNamespace)
+    val avscs = (srcDir ** AvroAvscFilter).get.map { avsc =>
+      log.info(s"Compiling Avro schemas $avsc")
+      new AvroFileRef(srcDir, avsc.relativeTo(srcDir).get.toString)
+    }
+    compileAvscs(avscs, target, stringType, fieldVisibility, enableDecimalLogicalType, useNamespace)
 
-    for (avpr <- (srcDir ** AvroAvrpFilter).get) {
-      log.info("Compiling Avro protocol %s".format(avpr))
+    (srcDir ** AvroAvrpFilter).get.foreach { avpr =>
+      log.info(s"Compiling Avro protocol $avpr")
       compileAvpr(avpr, target, stringType, fieldVisibility, enableDecimalLogicalType)
     }
 
-    (target ** "*.java").get.toSet
+    (target ** JavaFileFilter).get.toSet
```

Comment:
Generated java files are found with a file filter since the avro compiler doesn't return them.

We must generate those files in a separate target folder to avoid conflicts with other generators.